src/track/track.cpp

```diff
@@ -780,14 +779,41 @@ void Track::setCuePoints(const QList<CuePointer>& cuePoints) {
     for (const auto& pCue: m_cuePoints) {
         connect(pCue.get(), &Cue::updated, this, &Track::slotCueUpdated);
         // update main cue point
-        if (pCue->getType() == Cue::Type::MainCue) {
+        if (pCue->getType() == CueInfo::Type::MainCue) {
             m_record.setCuePoint(CuePosition(pCue->getPosition()));
         }
     }
     markDirtyAndUnlock(&lock);
     emit cuesUpdated();
 }
 
+void Track::setCuePoints(const QList<CueInfo>& cueInfos) {
+    QList<CuePointer> cuePoints;
+    const double sampleRateKhz = getSampleRate() / 1000.0;
```

Comment:
The conversion code could become a `Cue` constructor:

`Cue(const CueInfo& cueInfo, SINT sampleRateHz)`

Reply:
Good idea, but I think we need a `TrackId`, too. And there would be a performance penalty because I'd need to calculate the `millisecondsToSamplesFactor` for each individual cue point.

Reply:
I would not care about a single floating point devision that is invoked a few (< 10) times :rofl:

Reply:
Care about _allocations_, not _calculations_. Calculations only matter in very tight loops and time-critical situations.

Reply:
I could add the constructor and still keep this implementation, but I'm not sure if it makes sense to add a constructor if it's not used in the codebase.

Reply:
ok